tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,272 @@
+import { useBackend, useLocalState } from '../backend';
+import { Section, Flex, ProgressBar, Box, Button, Tabs, Stack, Input } from '../components';
+import { capitalize } from 'common/string';
+import { Window } from '../layouts';
+import { ElectricalPanel } from './common/ElectricalPanel';
+import { Fragment } from 'inferno';
+import { createLogger } from '../logging';
+
+export const Autolathe = (_props, context) => {
+  const { act, data } = useBackend(context);
+
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+    theme,
+  } = data;
+
+  return (
+    <Window width={600} height={600} theme={theme}>
+      <Window.Content scrollable>
+        <MaterialsData />
+        {queued && <QueueList />}
+        <PrintablesSection />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const MaterialsData = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+  } = data;
+
+  return (
+    <Section title="Materials">
+      <Flex direction="row" grow>
+        {Object.keys(materials).map((category) => (
+          <Flex.Item key={category} grow>
+            <ProgressBar
+              width="99%"
+              value={materials[category] / capacity[category]}>
+              <Box textAlign="center">
+                {capitalize(category)}: {materials[category]}/
+                {capacity[category]}
+              </Box>
+            </ProgressBar>
+          </Flex.Item>
+        ))}
+      </Flex>
+      <CurrentlyMaking />
+    </Section>
+  );
+};
+
+const CurrentlyMaking = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { currently_making } = data;
+
+  return currently_making ? (
+    <Fragment>
+      <Box height="5px" />
+      <Button
+        fluid
+        textAlign="center"
+        content={
+          'Currently making: ' +
```

Comment:
I'm not a huge fan of having decision making etc inside the jsx like this - if the conditional rendering above is used then this can be refactored out into a const and it'll be easier to read. You could also use string interpolation to simplify this